app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/MessageController.kt

```diff
@@ -18,28 +18,54 @@
 package com.infomaniak.mail.data.cache.mailboxContent
 
 import android.util.Log
+import com.infomaniak.lib.core.utils.monthsAgo
+import com.infomaniak.mail.data.api.ApiRepository
 import com.infomaniak.mail.data.cache.RealmDatabase
+import com.infomaniak.mail.data.models.Folder
+import com.infomaniak.mail.data.models.GetMessagesUidsDeltaResult.MessageFlags
 import com.infomaniak.mail.data.models.message.Message
 import com.infomaniak.mail.utils.copyListToRealm
+import com.infomaniak.mail.utils.toRealmInstant
 import io.realm.kotlin.MutableRealm
 import io.realm.kotlin.ext.query
+import io.realm.kotlin.ext.toRealmList
+import io.realm.kotlin.query.RealmQuery
+import io.realm.kotlin.query.RealmResults
 import io.realm.kotlin.query.RealmSingleQuery
+import kotlinx.coroutines.delay
+import java.text.SimpleDateFormat
+import java.util.*
+import kotlin.math.min
 
 object MessageController {
 
     //region Queries
+    private fun MutableRealm?.getMessagesQuery(uids: List<String>): RealmQuery<Message> {
+        val messages = "${Message::uid.name} IN {${uids.joinToString { "\"$it\"" }}}"
+        return (this ?: RealmDatabase.mailboxContent()).query(messages)
+    }
+
     private fun MutableRealm?.getMessageQuery(uid: String): RealmSingleQuery<Message> {
         return (this ?: RealmDatabase.mailboxContent()).query<Message>("${Message::uid.name} = '$uid'").first()
     }
```

Comment:
If the realm can be null, it's overkill to make an extention, because you use it like a variable with `this` after. So you might as well put it as a parameter.

Reply:
This will affect a lot of code, I'll do it in a dedicated PR.